File: src/go/svd/models/arrival_times.go
package models

import "time"

// ArrivalTimes represents the "Arrival Times" section of SVD.
type ArrivalTimes struct {
	Arrival                  time.Time `json:"arrival" xml:"Arrival" csv:"Arrival"`
	Departure                time.Time `json:"departure" xml:"Departure" csv:"Departure"`
	LocationEta              time.Time `json:"locationEta" xml:"LocationEta" csv:"LocationEta"`
	LocationActual           time.Time `json:"locationActual" xml:"LocationActual" csv:"LocationActual"`
	PilotBoardingPlaceEta    time.Time `json:"pilotBoardingPlaceEta" xml:"PilotBoardingPlaceEta" csv:"PilotBoardingPlaceEta"`
	PilotBoardingPlaceActual time.Time `json:"pilotBoardingPlaceActual" xml:"PilotBoardingPlaceActual" csv:"PilotBoardingPlaceActual"`
	VtsEta                   time.Time `json:"vtsEta" xml:"VtsEta" csv:"VtsEta"`
	VtsActual                time.Time `json:"vtsActual" xml:"VtsActual" csv:"VtsActual"`
	NextPortEta              time.Time `json:"nextPortEta" xml:"NextPortEta" csv:"NextPortEta"`
	VoyageTime               int       `json:"voyageTime" xml:"VoyageTime" csv:"VoyageTime"`
}

File: src/elixir/svd/lib/svd/models/arrival_times.ex
defmodule SVD.Models.ArrivalTimes do
  @moduledoc false

  @fields [
    :estimated_time_of_arrival,
    :estimated_time_of_completion,
    :estimated_time_of_pilot_boarding,
    :estimated_time_of_berth,
    :arrival,
    :departure,
    :location_eta,
    :location_actual,
    :pilot_boarding_place_eta,
    :pilot_boarding_place_actual,
    :vts_eta,
    :vts_actual,
    :next_port_eta,
    :voyage_time
  ]

  @derive {Jason.Encoder, only: @fields}
  defstruct @fields

  @type t :: %__MODULE__{}

  @spec fields() :: [atom()]
  def fields, do: @fields
end
